src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Process/Process.ts

```diff
@@ -55,6 +61,46 @@ export class Service {
     }
 }
 
+export var workflowSwitchDirective = (
+    adhConfig : AdhConfig.IService,
+    adhHttp : AdhHttp.Service<any>,
+    $window : angular.IWindowService
+) => {
+    return {
+        restrict: "E",
+        templateUrl: adhConfig.pkg_path + pkgLocation + "/WorkflowSwitch.html",
+        scope: {
+            path: "@"
+        },
+        transclude: true,
+        link: (scope, element) => {
+
+            adhHttp.options(scope.path, {importOptions: false}).then((raw) => {
+                scope.availableStates = AdhUtil.deepPluck(raw, [
+                    "data", "PUT", "request_body", "data", SIWorkflow.nick, "workflow_state"]);
+            });
```

Comment:
> Also, this won't update on login/logout. Maybe it would be good to extend adhPermissions.

Done in 6b89648a2e3bc959c6e6ddd0eb4acc325afdacda and 01f8f57f2450cb015a574ace1d42285cf1ee0b99.